lib/controllers/user/forgotPassword.controller.js

```diff
@@ -0,0 +1,158 @@
+"use strict";
+
+Object.defineProperty(exports, "__esModule", {
+  value: true
+});
+exports.resetPassword = exports.getResetPassword = exports.forgotPassword = void 0;
+var _jsonwebtoken = _interopRequireDefault(require("jsonwebtoken"));
+var _bcrypt = _interopRequireDefault(require("bcrypt"));
+var _models = _interopRequireDefault(require("../../database/models"));
+var _resetPasswordEmail = _interopRequireDefault(require("../../utils/resetPasswordEmail"));
+function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
+const User = _models.default.users;
+const resetSecret = process.env.RESET_SECRET;
+const forgotPassword = async (req, res) => {
+  const {
+    email
+  } = req.body;
+  try {
+    // look for email in database
+    const user = await User.findOne({
+      where: {
+        email
+      }
+    });
+    if (!user) {
+      res.status(404).json({
+        error: 'Invalid Email'
+      });
+    } else {
+      // res.status(200).json({message:'User exist'})
+
+      const token = _jsonwebtoken.default.sign({
+        user: user.email
+      }, resetSecret, {
+        expiresIn: '10m'
+      });
+      const link = `http://localhost:${process.env.PORT}/api/v1/user/reset-password/${user.id}/${token}`;
```

Comment:
The url should be defined in the env and passed as a variable

Reply:
We tried this, but this URL is containing varying parameters like **user.id** and **token**
In order for us to maintain a dynamic process of automatically capturing a unique **port** from .env, **user.id**, and **token** we figured out that embedding the URL like this would be the most convenient way.

Do you recommend adding such a varying URL in our env?

Reply:
The local host can change that can also be defined in the env

Reply:
Ooh yes, I have now made that adjustment.

Reply:
Ooh yes, I have now made that adjustment.